userspace/libsinsp/sinsp_filtercheck_thread.cpp

```diff
@@ -169,6 +170,25 @@ int32_t sinsp_filter_check_thread::extract_arg(string fldname, string val, OUT c
 			throw sinsp_exception("filter syntax error: " + val);
 		}
 	}
+	else if(m_field_id == TYPE_ENV ||
+			m_field_id == TYPE_AENV	)
+	{
+		if(val[fldname.size()] == '[')
+		{
+			size_t startpos = fldname.size();
+			parsed_len = (uint32_t)val.find(']', startpos);
+
+			if (startpos != std::string::npos && parsed_len != std::string::npos)
+			{
+				m_argname = val.substr(startpos + 1, parsed_len - startpos - 1);
+				parsed_len++;
+			}
```

Comment:
The same change proposed in the other PR of k8s fields, i would throw an exception here if we don't find `]`